### PR TITLE
Alibaba Cloud Linux(AliyunLinux) 2 and 3 kernel crawl

### PR DESCRIPTION
**Pull request by dark-vex**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area crawler

**What this PR does / why we need it**:
Add the capability to crawl kernels for Alibaba Cloud Linux (AliyunLinux) 2 and 3, it's a RHEL based distro, essentially  it's similar to AmazonLinux

**Test**
- AliyunLinux2
```
➜ docker run --rm --name kernel-test falcosecurity/kernel_crawler crawl --distro=AliyunLinux2 --version 4.19.91-26.al7
Checking repositories
Listing packages
{
  "AliyunLinux2": [
    {
      "kernelversion": 1,
      "kernelrelease": "4.19.91-26.al7.x86_64",
      "target": "aliyunlinux2",
      "headers": [
        "http://mirrors.aliyun.com/alinux/2/plus/x86_64/Packages/kernel-devel-4.19.91-26.al7.x86_64.rpm"
      ]
    }
  ]
}
```

- AliyunLinux3
```
➜ docker run --rm --name kernel-test falcosecurity/kernel_crawler crawl --distro=AliyunLinux3 --version 5.10.84-10.4.al8
Checking repositories
Listing packages
{
  "AliyunLinux3": [
    {
      "kernelversion": 1,
      "kernelrelease": "5.10.84-10.4.al8.x86_64",
      "target": "aliyunlinux3",
      "headers": [
        "http://mirrors.aliyun.com/alinux/3/plus/x86_64/Packages/kernel-devel-5.10.84-10.4.al8.x86_64.rpm"
      ]
    }
  ]
}
```

### CODE REVIEW

File: kernel_crawler/aliyunlinux.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            rpm.RpmMirror('http://mirrors.aliyun.com/alinux/', 'updates/' + arch + '/', v2_only),
+            rpm.RpmMirror('http://mirrors.aliyun.com/alinux/', 'plus/' + arch + '/', v2_only),
+
+        ]
+        super(AliyunLinux2Mirror, self).__init__(mirrors, arch)
+
+    def to_driverkit_config(self, release, deps):
+        for dep in deps:
+            if dep.find("devel") != -1:
+                return repo.DriverKitConfig(release, "aliyunlinux2", dep)
```
Comment (FedeDP) [inline kernel_crawler/aliyunlinux.py]: Is `aliyunlinux2` the os ID taken from `/etc/os-release`?

**Review by FedeDP — COMMENTED** [review]
PR lgtm!
I was just wondering whether the `DriverKitConfig` name matched the OS ID taken from `/etc/os-release`.
It is needed to eventually integrate this with Falco driver loader.
Thank you!

File: kernel_crawler/aliyunlinux.py (hunk 1 above)
Comment (dark-vex) [reply kernel_crawler/aliyunlinux.py]: on os-release it's `alinux` for both versions

Comment (FedeDP) [reply kernel_crawler/aliyunlinux.py]: So, can we merge both into a single distro and call the `DriverkitConfig`  `alinux`?

**Review by FedeDP — COMMENTED** [review]
LGTM!
Just a question: do we really want to split AliyunLinux{2,3}?
Cosindering they have the same OS_ID in /etc/os-release, we might want to have a single distro mapping them.

I mean, this is similar to what we do for CentOS.

**Comment by dark-vex** [comment]
@FedeDP we can but I noticed that the crawler become a bit slower this is why I have split them but if you prefer a single distro mapping I will change it

**Comment by FedeDP** [comment]
Yep i think a single mapping would be better :) 
Also, it's interesting: how could it be slower than the sum of crawling both of them? :thinking:

**Comment by dark-vex** [comment]
It's because the mirrors are slow, same happens with CentOS ones when you search a specific kernel

**Review by FedeDP — APPROVED** [review]
/approve
LGTM!

**Pull request merged**